src/App.js

```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+import 'bootstrap';
+import 'bootstrap/dist/css/bootstrap.min.css';
+
+import SetupPage from './SetupPage.js';
+import DeployPage from './DeployPage.js';
+import './App.css';
+
+export default class App extends React.Component {
+  render() {
+    let content = <SetupPage />;
+
+    if (window.location.search) {
```

Comment:
Perhaps this should "die". Perhaps, what this should do is, send the `deployments`, `owner`, and `repo` to the backend to get-or-create the shortlink and then redirect to that.